dialogflow-cx/agent_delegator.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Agent Delegator module. Coordinates agent state with Dialogflow."""
+
+
+import client_delegator as cd
+import dialogflow_sample as ds
+import google.api_core.exceptions
+import google.auth
+from google.cloud.dialogflowcx import (
+    Agent,
+    AgentsClient,
+    DeleteAgentRequest,
+    GetAgentRequest,
+    ListAgentsRequest,
+)
+
+
+class AgentDelegator(cd.ClientDelegator):
+    """Class for organizing interactions with the Dialogflow Agent API."""
+
+    _DEFAULT_LANGUAGE_CODE = "en"
+    _DEFAULT_TIME_ZONE = "America/Los_Angeles"
+    _CLIENT_CLASS = AgentsClient
+
+    def __init__(self, controller: ds.DialogflowSample, **kwargs) -> None:
+        super().__init__(controller, **kwargs)
+        self._agent = None
+        self.default_language_code = kwargs.get(
+            "default_language_code", self._DEFAULT_LANGUAGE_CODE
+        )
+        self.time_zone = kwargs.get("time_zone", self._DEFAULT_TIME_ZONE)
+
+    def initialize(self):
```

Comment:
(This comment applies to a few classes) I wonder if the name should be adjusted a bit to disambiguate between this and the standard __init__() method.

Reply:
Changing to "setup".